Projects/Offer/Sources/Components/Header/DiscountTag.swift

```diff
@@ -0,0 +1,97 @@
+        horizontalCenteringStackView.edgeInsets = UIEdgeInsets(inset: 10)
+        horizontalCenteringStackView.axis = .vertical
+        horizontalCenteringStackView.alignment = .center
+        horizontalCenteringStackView.distribution = .equalCentering
+        view.addSubview(horizontalCenteringStackView)
+        
+        horizontalCenteringStackView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+        
+        let contentStackView = UIStackView()
+        contentStackView.axis = .horizontal
+        contentStackView.spacing = 2
+        contentStackView.alignment = .center
+        contentStackView.distribution = .equalCentering
+        horizontalCenteringStackView.addArrangedSubview(contentStackView)
+        
+        let textStyle = TextStyle.brand(.caption1(color: .primary(state: .positive))).centerAligned.uppercased
+        
+        let titleLabel = UILabel(
+            value: "",
+            style: textStyle
+        )
+        contentStackView.addArrangedSubview(titleLabel)
+
+        let subtitleLabel = UILabel(
+            value: "",
+            style: textStyle
+        )
+        contentStackView.addArrangedSubview(subtitleLabel)
+        
+        bag += state.dataSignal.map { $0.redeemedCampaigns.first }.onValue { campaign in
+            guard let campaign = campaign else {
+                view.isHidden = true
+                return
+            }
+            
+            view.isHidden = false
+            
+            let incentiveFragment = campaign.fragments.campaignFragment.incentive?.fragments.incentiveFragment
+
+            if let freeMonths = incentiveFragment?.asFreeMonths {
+                titleLabel.value = L10n.offerScreenFreeMonthsBubbleTitle
+                titleLabel.animationSafeIsHidden = false
+                subtitleLabel.value = L10n.offerScreenFreeMonthsBubble(freeMonths.quantity ?? 0)
+            } else if let percentageDiscount = incentiveFragment?.asPercentageDiscountMonths {
+                titleLabel.value = L10n.offerScreenPercentageDiscountBubbleTitle
+                titleLabel.animationSafeIsHidden = false
+                let months = percentageDiscount.percentageNumberOfMonths
+                if months == 1 {
+                    subtitleLabel.value = L10n.offerScreenPercentageDiscountBubbleTitleSingular(Int(percentageDiscount.percentageDiscount))
+                } else {
+                    subtitleLabel.value = L10n.offerScreenPercentageDiscountBubbleTitlePlural(Int(percentageDiscount.percentageDiscount), months)
+                }
+            } else if let monthlyDeduction = incentiveFragment?.asMonthlyCostDeduction {
+                titleLabel.value = L10n.offerScreenPercentageDiscountBubbleTitle
+                titleLabel.animationSafeIsHidden = false
+                subtitleLabel.value = "-\(monthlyDeduction.amount?.fragments.monetaryAmountFragment.monetaryAmount.formattedAmount ?? "")\(L10n.perMonth)"
```

Comment:
maybe `monthlyDeduction.amount?.fragments.monetaryAmountFragment.monetaryAmount.negative.formattedAmount ?? ""` is a bit cleaner? also we could add a `prefix func -` to monetary amount, I think the negative is good enough though.

Reply:
![Screenshot 2021-05-03 at 09 55 36](https://user-images.githubusercontent.com/23550083/116841104-bbbd1380-abf5-11eb-83df-0de363d07553.png)

Reply:
`Coverage` has some hardcoded strings.

Reply:
Ah coverage ain't done yet. Is there a ticket for this section?

Reply:
@tarikstafford will actually replace this logic with this https://github.com/HedvigInsurance/giraffe/pull/1169 too much damn code to unwrap these 🤠 & yeah, there's a bit of untranslated stuff right now

Reply:
@tarikstafford updated now!